docs/guide/backend-integration.md

```diff
@@ -85,10 +87,30 @@ If you need a custom integration, you can follow the steps in this guide to conf
    - For non entry chunks, the key is the base name of the generated file prefixed with `_`.
    - Chunks will contain information on its static and dynamic imports (both are keys that map to the corresponding chunk in the manifest), and also its corresponding CSS and asset files (if any).
 
-   You can use this file to render links or preload directives with hashed filenames (note: the syntax here is for explanation only, substitute with your server templating language):
+4. A backend generating HTML should include the following tags given an entry point path:
+
+   - A tag for the `file` key of the entry point chunk (`<script type="moudle">` for Javascript, or `<link rel="stylesheet">` for css)
+   - A `<link rel="stylesheet">` tag for each file in the entry point chunk's `css` list
+   - Recursively, a `<link rel="stylesheet">` tag for the each chunk referenced in the `imports` list
+   - Optionally, `<link rel="modulepreload">` tag for each Javascript chunk referenced by the imports list, again recursively following the imports of each chunk
+
+   Following the above example manifest, for the entry point `main.js` the following tags should be included in production:
+
+   ```html
+   <!-- if production -->
+   <link rel="stylesheet" href="assets/main.b82dbe22.css" />
+   <link rel="stylesheet" href="assets/shared.a834bfc3.css" />
+   <script type="module" src="assets/main.4889e940.js"></script>
+   <!-- optional -->
+   <link rel="modulepreload" src="assets/shared.83069a53.js" />
```

Comment:
I think it is still useful to first leave the template based one. And then you can add the concrete examples:

You can use this file to render links or preload directives with hashed filenames (note: the syntax here is for explanation only, substitute with your server templating language. The `cssDeps` and `jsDeps` are for illustration and aren't provided by Vite):
```
   <!-- if production -->

   <link rel="stylesheet" href="/{{ manifest[name].css }}" />
   <!-- for chunk of cssDeps(manifest[name]) -->
     <link rel="stylesheet" href="/{{ chunk.css }}" }/>

   <script type="module" src="/{{ manifest[name].file }}"></script>
   <!-- for chunk of jsDeps(manifest[name]), optional -->
     <link rel="modulepreload" src="/{{ chunk.file }}" />
```

Maybe we should expose at one point some utils to work with the manifest, like `jsDeps` and `cssDeps`.